src/services/sanity.js

```diff
@@ -1249,21 +1250,27 @@ export async function fetchGenreLessons(brand, name, contentType, {
 }
 
 
-
 /**
- * Fetch data from the Sanity API based on a provided query.
  *
  * @param {string} query - The GROQ query to execute against the Sanity API.
  * @param {boolean} isList - Whether to return an array or a single result.
- * @returns {Promise<Object|null>} - A promise that resolves to the fetched data or null if an error occurs or no results are found.
+ * @param {Function} [customPostProcess=null] - custom post process callback
+ * @param {boolean} [processNeedAccess=true] - execute the needs_access callback
+ * @returns {Promise<*|null>} - A promise that resolves to the fetched data or null if an error occurs or no results are found.
  *
  * @example
  * const query = `*[_type == "song"]{title, artist->name}`;
  * fetchSanity(query, true)
  *   .then(data => console.log(data))
  *   .catch(error => console.error(error));
  */
-export async function fetchSanity(query, isList) {
+
+export async function fetchSanity(query,
+                                  isList,
+                                  { customPostProcess = null,
+                                    processNeedAccess = true,} = {}
```

Comment:
I'm not a big fan of this style, to be honest, but I really like the way you can leverage it to use named optional params, so well done there. I think it just highlights how TypeScript is superior to JavaScript 😄

Reply:
Yea, I'm in the same boat. It's a workaround for a way that other languages just handle nicely.